libs/ui-components/src/components/Repository/CreateRepository/utils.ts

```diff
@@ -278,23 +282,25 @@ export const getRepositoryPatches = (values: RepositoryFormValues, repository: R
 };
 
 const gitRegex = new RegExp(/^((http|git|ssh|http(s)|file|\/?)|(git@[\w.]+))(:(\/\/)?)([\w.@:/\-~]+)(\.git)?(\/)?$/);
-const repoNameRegex = /^[a-zA-Z0-9-_\\.]+$/;
 const pathRegex = /\/.+/;
 
 export const repoSyncSchema = (t: TFunction, values: ResourceSyncFormValue[]) => {
   return Yup.array()
     .of(
       Yup.object().shape({
-        name: Yup.string()
-          .defined(t('Name is required.'))
-          .test('Must be unique', t('Must be unique'), (value) => {
-            if (!value) {
-              return true;
-            }
-            return values.filter((v) => v.name === value).length === 1;
-          }),
-        targetRevision: Yup.string().defined(t('Target revision is required.')),
-        path: Yup.string().matches(pathRegex, t('Must be an absolute path.')).defined(t('Path is required.')),
+        name: validKubernetesLabel(t, { isRequired: true }).test('Must be unique', t('Must be unique'), (value) => {
```

Comment:
we should have a name-specific validation. The valid values are a bit different - 
 we should link https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names (dns subdomain is also used by fctl backend.

Reply:
I see. I suggest to continue with those validations in the follow-up for the interactive checker.

With this PR, we'd make sure that the UI won't have 400 errors, and I'll continue handling the the more nuanced validations for labels there.

Reply:
Ok, lets improve in a follow-up